sql/analyzer/pushdown.go

```diff
@@ -672,6 +685,19 @@ func pushdownFiltersUnderSubqueryAlias(ctx *sql.Context, a *Analyzer, sa *plan.S
 // pushdownIndexesToTable attempts to convert filter predicates to indexes on tables that implement
 // sql.IndexAddressableTable
 func pushdownIndexesToTable(a *Analyzer, tableNode sql.NameableNode, indexes map[string]*indexLookup) (sql.Node, transform.TreeIdentity, error) {
+	hasLimit := false
```

Comment:
this one confuses me a bit. This function is called on `*plan.ResolvedTable` and `*plan.TableAlias`. Can either of those have a limit as a child?